WalletWasabi/Blockchain/Transactions/TransactionStore.cs

```diff
@@ -35,23 +36,25 @@ public async Task InitializeAsync(string workFolderPath, Network network, string
 
 				var fileName = Path.Combine(WorkFolderPath, "Transactions.dat");
 				var transactionsFilePath = Path.Combine(WorkFolderPath, fileName);
-				TransactionsFileManager = new MutexIoManager(transactionsFilePath);
 
-				using (await TransactionsFileManager.Mutex.LockAsync().ConfigureAwait(false))
+				// In Transactions.dat every line starts with the tx id, so the first character is the best for digest creation.
+				TransactionsFileManager = new DigestableIoManager(transactionsFilePath, 0);
```

Comment:
Why did `MutexIoManager` became `DigestableIoManager`?

Reply:
- DigestableIoManager inherited from SafeIoManager that is needed for us to be able to recover from corrupted IO operation. 
- DigestableIoManager capable of detecting if two operations writing the same content to the file and skip the second write. 
- For the sake of unified procedures DigestableIoManager is also used by FilterStore in a similar manner.

Reply:
We agreed on devmeeting to not change the business object here in this PR, so I am using the same logic behind that is IoManager and add Asynclock externally.